File: src/python/jterator/api/tools.py
import plotly.plotly as py
import matplotlib.pyplot as plt


def jtfigure(fig, filename, fig_format):
    if fig_format == 'pdf':
        plt.savefig('figures/%s.pdf' % filename, format='pdf')
        plt.close()
    elif fig_format == 'plotly':
        figure_url = py.plot_mpl(fig, filename=filename)
        return figure_url

File: src/r/jterator/tools.R
library(plotly)
library(ggplot)


jtfigure <- function(fig, filename, fig_format) {
    if (fig_format == 'pdf') {
        pdf(sprintf("figures/%s.pdf", filename))
        print(fig)
        dev.off()
    }
    else if (fig_format == 'plotly') {
        py <- plotly()
        figure_url <- py$ggplotly()
        return(figure_url)
    }
}
